app/src/main/java/io/intrepid/contest/screens/contestoverview/DualCategoryScoreAdapter.java

```diff
@@ -0,0 +1,69 @@
+package io.intrepid.contest.screens.contestoverview;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.ViewGroup;
+
+import java.util.List;
+
+import io.intrepid.contest.models.Category;
+import io.intrepid.contest.models.ScoreWeight;
+import io.intrepid.contest.screens.contestcreation.categorieslist.CategoryViewHolder;
+
+class DualCategoryScoreAdapter extends RecyclerView.Adapter {
+    private final int CATEGORY_VIEW_TYPE = 111;
+    private final int SCORE_VIEW_TYPE = 222;
+    private List<Category> categories;
+    private List<ScoreWeight> scoreWeights;
```

Comment:
Thanks @kentonguan Is this in general how the adapters should be set up or are you recommending it in this particular case just to avoid the null checking